src/compiler/crystal/codegen/codegen.cr

```diff
@@ -592,14 +592,7 @@ module Crystal
 
       if obj = node.obj
         accept obj
-
-        # If obj is a primitive like an integer we need to pass
-        # the variable as is (without loading it)
-        if obj.is_a?(Var) && obj.type.is_a?(PrimitiveType)
-          call_self = context.vars[obj.name].pointer
-        else
-          call_self = @last
-        end
+        call_self = @last
```

Comment:
Why is the "If obj is a primitive like ..." no longer needed?

Reply:
Because if `obj` is a `Var` thenh this was turned into a `ProcLiteral`, so a proc pointer at this point **never** has a `Var` as its `obj`.